news/bugfix-3188.md

```diff
@@ -0,0 +1,3 @@
+afmongodb: do not build module when ENABLE_MONGODB=OFF
+
+fixes #3184
```

Comment:
As this change does not fix #3184, please remove this line, and also the `fixes` line from the PR description. Apart from that, the change looks good to me. I agree this change is necessary, but we still need to find a way to deal with #3184

Reply:
Now, it fixes...

But I still disagree that we install a package that has a bug - as you want to disable the module that uses this package, why do we install it?
There are two possible solutions:
* install a package that contains the fix (from newest distro, if they fixed; or from OBS if no other choice)
 * do not install the package... 

Workarounding packaging bugs in this way is not the best, IMHO.

Reply:
I agree. It would be preferable to install the patched package, especially after going through fixing it in obs. It is a half solution, because in the end, we will not test mongodb, which would be a goal for a ci job.

On the other hand, in general: if a module is disabled, it would be nice if cmake does not run detection for it, because it is unnecessary. Independently from this problem. Maybe `module_switch` might be improved that way, so that could be applied to all modules.

 We use autotools in that way. (At least in case of riemann which I quicky checked, there may be exceptions :))

```bash
if test "$enable_riemann" != "no"; then
 PKG_CHECK_MODULES(RIEMANN_CLIENT, riemann-client >= $LRC_MIN_VERSION,,riemann_found="no")
 if test "$riemann_found" = "no" && test "$enable_riemann" = "yes"; then
   AC_MSG_ERROR([Dependency for Riemann not found!])
 fi
 if test "$riemann_found" = "no"; then
   enable_riemann="no";
 else
   enable_riemann="yes";
 fi

 if test "$enable_riemann" = "yes"; then
```

With all that said, I would have been fine if we merged this PR with its previous version, find a better way to deal with mongodb, and leave the `disable->do-not-run-detection` as future improvement. But the current version is also fine, just we may look back later to do the generalization.